File: day-1/day-1.hs
main = do
  contents <- readFile "input.txt"
  print (solve (depths contents) 0 0)
  print (solve (depths contents) 2 0)

depths :: [Char] -> [Int]
depths contents = map read (lines contents)

solve :: [Int] -> Int -> Int -> Int
solve (x:xs) idx count =
  if length xs == idx
  then count
  else
    solve xs idx (count + countIncrease (x < (xs !! idx)))

countIncrease :: Bool -> Int
countIncrease True = 1
countIncrease False = 0

File: day-1/day-1.c
#include <assert.h>
#include <stdio.h>
#include <stdlib.h>

#define INPUTLENGTH 2000

void get_input(int *input) {
  FILE *fp = fopen("./input.txt", "r");

  char *line = NULL;
  size_t len = 0;

  for (int i = 0; getline(&line, &len, fp) != -1; i += 1) {
    input[i] = atoi(line);
  }

  fclose(fp);
}

int solve(int *input, int n) {
  int count = 0;
  for (int i = n; i < INPUTLENGTH; i += 1) {
    if (input[i] > input[i - n]) {
      count += 1;
    }
  }
  return count;
}

int main() {
  int input[INPUTLENGTH];
  get_input(input);

  assert(solve(input, 1) == 1832);
  assert(solve(input, 3) == 1858);

  return 0;
}
